### PR TITLE
feat(circleci): suppourt basic function

### PR DESCRIPTION
**Pull request by soulbird**
### Description

feat(circleci): suppourt basic function

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: circleci/pod/docker-compose.common.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+version: "3.8"
+
+services:
+  ## Etcd
+  etcd_old:
+    image: rancher/coreos-etcd:v3.4.16-arm64
```
Comment (spacewander) [inline circleci/pod/docker-compose.common.yml]: The old etcd requires to use an etcd which is too old to support

File: utils/linux-install-etcd-client.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
```
Comment (membphis) [inline utils/linux-install-etcd-client.sh]: can we use a new shell script? we do not want to modify any file from APISIX project.

File: utils/linux-install-openresty.sh [hunk 3]
```diff
@@ -17,10 +17,16 @@
 #
 set -euo pipefail
 
+ARCH=${ARCH:-`(uname -m | tr '[:upper:]' '[:lower:]')`}
```
Comment (membphis) [inline utils/linux-install-openresty.sh]: ditto

**Review by membphis — COMMENTED** [review]
and we need a good title for this PR

**Review by membphis — APPROVED** [review]
LGTM

**Pull request merged**